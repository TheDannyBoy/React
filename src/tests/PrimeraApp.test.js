// import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import PrimeraApp from '../components/PrimeraApp';

describe('Testing - PrimeraApp component', () => {
  test('Should show <PrimeraApp /> correctly', () => {
    const saludo = 'Hola, soy Goku';
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Should show the subtitle sent from the props', () => {
    const saludo = 'Hola, soy Goku';
    const subtitulo = 'Soy un subtitulo';
    const wrapper = shallow(<PrimeraApp
      saludo={saludo}
      subtitulo={subtitulo}
    />);

    const textoParrafo = wrapper.find('p').text();
    expect(textoParrafo).toBe(subtitulo);
  });
});
