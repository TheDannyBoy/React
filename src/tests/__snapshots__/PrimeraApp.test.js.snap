// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing - PrimeraApp component Should show <PrimeraApp /> correctly 1`] = `
<Fragment>
  <h1>
    Hola, soy Goku
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
